Darwin/Services/StitchRemoteMongoDBService/StitchRemoteMongoDBServiceTests/SyncPerformanceContext.swift

```diff
@@ -0,0 +1,232 @@
+import MongoSwift
+import StitchCore
+import StitchCoreSDK
+import StitchCoreAdminClient
+import StitchDarwinCoreTestUtils
+@testable import StitchCoreRemoteMongoDBService
+import StitchCoreLocalMongoDBService
+@testable import StitchRemoteMongoDBService
+
+protocol SyncPerformanceContext {
+    var dbName: String { get }
+    var collName: String { get }
+    var userId: String { get }
+    var client: StitchAppClient { get }
+    var mongoClient: RemoteMongoClient { get }
+    var coll: RemoteMongoCollection<Document> { get }
+    var testParams: TestParams { get }
+    var transport: FoundationInstrumentedHTTPTransport { get }
+    var harness: SyncPerformanceIntTestHarness { get }
+    var streamJoiner: StreamJoiner { get }
+    var joiner: ThrowingCallbackJoiner { get }
+
+    init(harness: SyncPerformanceIntTestHarness,
+         testParams: TestParams,
+         transport: FoundationInstrumentedHTTPTransport) throws
+
+    func tearDown() throws
+}
+
+extension SyncPerformanceContext {
+    func runSingleIteration(numDocs: Int,
+                            docSize: Int,
+                            testDefinition: TestDefinition,
+                            setup: SetupDefinition,
```

Comment:
[naming] let's call these `customSetup` and `customTeardown` so it's more clear what they do

Reply:
I disagree with that. `custom` is implied.

Reply:
lol, jason had me take the word custom out....

Reply:
alright fair enough. I still think we should name them `beforeEach` and `afterEach` in the other class though